src/NServiceBus.AcceptanceTests/Routing/AutomaticSubscriptions/When_handling_local_event.cs

```diff
@@ -0,0 +1,79 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing.AutomaticSubscriptions
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_handling_local_event : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_autosubscribe_to_event()
+        {
+            var ctx = await Scenario.Define<Context>(x => x.Id = Guid.NewGuid())
+                .WithEndpoint<PublisherAndSubscriber>(b => b
+                .When((session, context) =>
+                {
+                    if (context.HasNativePubSubSupport)
+                    {
+                        context.EventSubscribed = true;
+                    }
+                    return Task.FromResult(0);
+                })
+                .When(c => c.EventSubscribed || c.HasNativePubSubSupport, (session, context) => session.Publish(new Event { ContextId = context.Id })))
+                .Done(c => c.GotEvent)
+                .Run().ConfigureAwait(false);
+
+            Assert.True(ctx.GotEvent);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public Guid Id { get; set; }
+            public bool GotEvent { get; set; }
+            public bool EventSubscribed { get; set; }
+        }
+
+        public class PublisherAndSubscriber : EndpointConfigurationBuilder
+        {
+            public PublisherAndSubscriber()
+            {
+                EndpointSetup<DefaultPublisher>(b =>
+                {
+                    // Make sure the subscription message isn't purged on startup
+                    b.PurgeOnStartup(true);
```

Comment:
>So I think there should be a test ensuring that PurgeOnStartup does not remove local sent messages?

To my knowledge this would be impossible to support in ASQ? (cc @Particular/azure-storage-queues-maintainers )

My longterm proposal would be to deprecate the purge feature completely and add guidance how to purge using "scripting" like eg. http://docs.particular.net/nservicebus/msmq/operations-scripting since purging is a very dangerous operation.